src/Message/CompleteAuthoriseAndCaptureResponse.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Omnipay\IcepayPayments\Message;
+
+/**
+ * The response after complete authorise and capture request.
+ * For this response, we explicitly check what the status is of the payment transaction at Icepay.
+ */
+class CompleteAuthoriseAndCaptureResponse extends AbstractResponse
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isSuccessful(): bool
+    {
+        return isset($this->data['status']) && in_array($this->data['status'], [
+            self::RESPONSE_STATUS_COMPLETED,
+            self::RESPONSE_STATUS_SETTLED,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * In case there is no status 'cancelled' available yet, look up for a statusCode queryString in the HttpRequest.
+     * Icepay redirects you to the complete page with the statusCode query string.
+     */
+    public function isCancelled(): bool
+    {
+        return isset($this->data['status']) && $this->data['status'] === self::RESPONSE_STATUS_CANCELLED
+            || $this->getRequest()->getHttpRequest()->get('statusCode') === self::RESPONSE_STATUS_CANCELLED;
```

Comment:
What would happen if `statusCode` is not set in the current request?

Reply:
It has two options. Either the response status has been set or queryString has statusCode. If one of them has the value of `cancelled` it will return true, else it will return false and the payment transaction cannot be set as cancelled.

Reply:
Looked it up myself. The get method will return NULL.